5-performance-optimization.ipynb

```diff
@@ -0,0 +1,764 @@
+   "id": "f06a2f61-89b5-4a1f-9928-da4fe14d62a5",
+   "metadata": {},
+   "source": [
+    "<img src=\"images/dask_horizontal.svg\" width=\"45%\" alt=\"Dask logo\\\">\n",
+    "\n",
+    "# Optimizing Dask Workloads\n",
+    "\n",
+    "This notebook illustrates a common Dask ETL workload, and demonstrates how one might go about diagnosing and resolving performance issues using the dashboard. We will provide a motivating example for where a user may require Dask, and outline some common do's and don't's of Dask dataframe operations.\n",
+    "\n",
+    "## Cluster setup\n",
+    "\n",
+    "Because the data we are using resides in AWS S3, we will be spinning up a [Coiled cluster](https://docs.coiled.io/user_guide/clusters/index.html) in the same region to minimize I/O costs (is it worth adding a blurb that some of the presenters are Coiled employees?):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "29f3298c-6626-4eb4-8389-1e3ca7f83f67",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import coiled\n",
+    "\n",
+    "cluster = coiled.Cluster(n_workers=50, region=\"us-east-2\")  # start workers close to data to minimize costs\n",
+    "client = cluster.get_client()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b3396b6a-6432-4e86-b826-17556efee8ec",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "client"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "2a4c4623-e950-4685-849c-9793b5f18e59",
+   "metadata": {},
+   "source": [
+    "## Motivation: TLC Trip Records\n",
+    "\n",
+    "The New York City Taxi and Limousine Commission (TLC) collects trip record information for each taxi and for-hire vehicle trip completed by licensed drivers and vehicles; a subset of this data (~60 GB) stored on S3 provides a good example of an out-of-core dataset that we would otherwise be unable to explore on a standard laptop.\n",
+    "\n",
+    "Using `dask.dataframe.read_csv`, we can lazily read this data in and do some low-level exploration before performing more complex computations:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3b486301-0df1-4700-b618-343bbb397013",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "\n",
+    "import dask.dataframe as dd\n",
+    "\n",
+    "ddf = dd.read_csv(\n",
+    "    \"s3://coiled-datasets/uber-lyft-tlc-sample/csv-10/*\", \n",
+    "    # \"s3://coiled-datasets/uber-lyft-tlc-sample/csv-ill/*\", \n",
+    "    dtype={\"wav_match_flag\": \"category\"},  # worth adding a blurb around why we set this?\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "da2e59c0",
+   "metadata": {},
+   "source": [
+    "After some initial exploration, we see that the columns representing on-scene and pickup times are stored as strings(?)"
```

Comment:
It would be good to add that we are also creating some new columns to offer different views into the data.